### ISSUE TITLE
Configurable overpower protection

### ISSUE THREAD

**Issue by KrzysztofHajdamowicz** [bieniu/ha-shellies-discovery-gen2#165]
**Is your feature request related to a problem? Please describe.**
I am using Shelly Plus Plug S in a remote location, with a little flaky internet connection.  
Role of this plug is to measure and control battery charger.
Currently, I have Home Assistant automation, that turns off smart plug outlet when chargers are consuming more than X Watts of power (e.g. approaching 80% state of battery charge).

I'd like to offload this functionality to smart plug itself (so outlet can be turned off by plug itself).

**Describe the solution you'd like**

I'd like to see switch `Overpower protection` and input_number (number to enter, maybe slider) to specify amount of power `device connected to outlet` is allowed to draw.

**Additional context**
According to documentation,
https://shelly-api-docs.shelly.cloud/gen2/ComponentsAndServices/Switch
There is `power_limit` attribute possible to configure.

**Comment by bieniu** [comment]
As long as the device doesn't periodically send data from `Switch.GetConfig` payload, we are unable to implement such a feature. MQTT integration is based on data automatically sent by the device, it doesn't poll the device. Shelly for Gen2 devices takes a different approach, mostly returning data by polling only.

**Comment by KrzysztofHajdamowicz** [comment]
Can't we call `Switch.GetConfig()` from javascript?
https://github.com/bieniu/ha-shellies-discovery-gen2/blob/master/scripts/send_device_status.js

**Comment by KrzysztofHajdamowicz** [comment]
Rationale: In case of flaky connection, it's better to have a lag on _setting overpower value and fast overpower reaction_ than _having a lag in both getting overpower data and reacting to it_

**Comment by bieniu** [comment]
> Can't we call `Switch.GetConfig()` from javascript?

Yes, we can but… we should call GetConfig for every relay (we need information how many relays exist) and merge the payload with the original status payload. Not a pleasant task in js.

BTW You can configure the overpower value on the device's web panel and the relay will turn off when an overpower event occurs.

**Comment by KrzysztofHajdamowicz** [comment]
> BTW You can configure the overpower value on the device's web panel and the relay will turn off when an overpower event occurs.

Yes, that's a good workaround in static environment... Sometimes I charge battery using 2x5A chargers, sometimes 1x5A, also 1x3,5 and 2x3,5A. Also happens to use charger for other battery voltage. It's dynamic.

I wanted to make some automation that will discover (at least some of) used device and adjust overpower protection.

> we should call GetConfig for every relay (we need information how many relays exist) and merge the payload with the original status payload. Not a pleasant task in js.

How about calling GetConfig blindly for each possible relay (0 to 3?) and sending result to different MQTT topics, then merging it on python-sript-side?

**Comment by bieniu** [comment]
> How about calling GetConfig blindly for each possible relay (0 to 3?) and sending result to different MQTT topics, then merging it on python-sript-side?

Not possible, this would require a change in MQTT integration code.

IMO you should configure overpower switch and number entities by yourself with an old good yaml 😁
